src/machine/state/sr.rs

```diff
@@ -25,23 +27,26 @@ impl Into<u8> for R {
     }
 }
 
-pub struct SRegisters(());
+
+pub struct SRegisters(Vec<Octa>);
 
 impl SRegisters {
     pub fn new() -> Self {
-        unimplemented!();
+        SRegisters(vec![0u64.into(); 32])
     }
 }
 
 impl Index<R> for SRegisters {
     type Output = Octa;
-    fn index(&self, _: R) -> &Self::Output {
-        unimplemented!();
+    fn index(&self, index: R) -> &Self::Output {
+        let i: u8 = index.into();
+        self.0.index(i as usize)
```

Comment:
The second proposal does not compile.
I thought that as we implemented .into() for u8 we might as well use it.